test/Microsoft.Azure.Devices.Edge.Test.Common/linux/OsPlatform.cs

```diff
@@ -43,8 +45,18 @@ public async Task<CaCertificates> GenerateCaCertificatesAsync(string deviceId, s
         public CaCertificates GetEdgeQuickstartCertificates() =>
             this.GetEdgeQuickstartCertificates("/var/lib/iotedge/hsm");
 
-        public void InstallCaCertificates(IEnumerable<X509Certificate2> certs, ITransportSettings _) =>
-            this.InstallTrustedCertificates(certs, StoreName.Root);
+        public void InstallCaCertificates(IEnumerable<X509Certificate2> certs, ITransportSettings transportSettings)
+        {
+            if (transportSettings.GetTransportType() == TransportType.Amqp_WebSocket_Only ||
+                transportSettings.GetTransportType() == TransportType.Amqp_Tcp_Only)
+            {
+                transportSettings.SetupCertificateValidation(certs.First());
```

Comment:
Why do we need diff behavior for AMQP?

Reply:
this is only an issue with AMQP.  For MQTT, I think it's better to preserve the behavior of installing the CA cert, so that our test reflects the customer scenarios.

---
In reply to: [373151134](https://github.com/Azure/iotedge/pull/2436#discussion_r373151134) [](ancestors = 373151134)